omc3/model/accelerators/lhc.py

```diff
@@ -486,12 +488,42 @@ def _get_madx_main_sequence_loading(self) -> str:
                 "has to be specified (--lhcmode option missing?)."
             )
 
-    def get_update_correction_script(self, outpath: Path | str, corr_files: Sequence[Path | str]) -> str:
+    def get_update_correction_script(self, outpath: Path | str, corr_files: Sequence[Path | str], variable_categories: Sequence[str]) -> str:
         madx_script = self.get_base_madx_script()
         for corr_file in corr_files:
             madx_script += f"call, file = '{str(corr_file)}';\n"
-        madx_script += f"exec, do_twiss_elements(LHCB{self.beam}, '{str(outpath)}', {self.dpp});\n"
+        
+        if DELTAP_NAME in variable_categories: # If the dpp is a variable in the corrections, do twiss, correct, match
```

Comment:
same comment about the `DELTAP_NAME` in the categories/variables list here. BUT as you do not need all of the variables in this function (and also hopefully never will) I would simply pass a boolean i.e. `update_dpp: bool = False` as a parameter.